docs/content/tutorial/step_04.ngdoc

```diff
@@ -1,199 +1,310 @@
+
+* Put each entity in its **own file**.
+* Organize our code by **feature area**, instead of by function.
+* Split our code into **modules** that other modules can depend on.
+
+<div class="alert alert-info">
+  We will keep it short, not going into great detail on every good practice and convention. These
+  principles are explained in great detail in the [Angular Style Guide][styleguide], which also
+  contains many more techniques for effectively organizing Angular codebases.
+</div>
 
 
 <div doc-tutorial-reset="4"></div>
 
 
-## Template
+## One Feature per File
 
-__`app/index.html`:__
+It might be tempting, for the sake of simplicity, to put everything in one file, or have one file
+per type; e.g. all controllers is one file, all components in another file, all services in a third
+file, and so on.
+This might seem to work well in the beginning, but as our application grows it becomes a burden to
+maintain. As we add more and more features, our files will get bigger and bigger and it will be
+difficult to navigate and find the code we are looking for.
 
-```html
-  Search: <input ng-model="query">
-  Sort by:
-  <select ng-model="orderProp">
-    <option value="name">Alphabetical</option>
-    <option value="age">Newest</option>
-  </select>
+Instead we should put each feature/entity in its own file. Each stand-alone controller will be
+defined in its own file, each component will be defined in each own file, etc.
 
+Luckily, we don't need to change anything with respect to that guideline in our code, since we have
+already defined our `phoneList` component in its own `phone-list.component.js` file. Good job!
+
+We will keep this in mind though, as we add more features.
 
-  <ul class="phones">
-    <li ng-repeat="phone in phones | filter:query | orderBy:orderProp">
-      <span>{{phone.name}}</span>
-      <p>{{phone.snippet}}</p>
-    </li>
-  </ul>
-```
 
-We made the following changes to the `index.html` template:
+## Organizing by Feature
 
-* First, we added a `<select>` html element named `orderProp`, so that our users can pick from the
-two provided sorting options.
+So, now that we learned we should put everything in its own file, our `app/` directory will soon be
+full with dozens of files and specs (remember we keep our unit test files next to the corresponding
+source code files). What's more important, logically related files will not be groupped together; it
+will be really difficult of locate all files related to a specific section of the application and
+make a change or fix a bug.
 
-<img class="diagram" src="img/tutorial/tutorial_04.png">
+So, what shall we do?
 
-* We then chained the `filter` filter with {@link ng.filter:orderBy `orderBy`}
-filter to further process the input into the repeater. `orderBy` is a filter that takes an input
-array, copies it and reorders the copy which is then returned.
+Well, we are going to group our files into directories _by feature_. For example, since we have a
```

Comment:
Hm...based on (unofficial) resources like [this][1], [this][2] or [this][3], I'd say it's definitely not a Windows thing - or at least that `folder` is definitely a Windows thing :stuck_out_tongue: 
In addition, based on the fact that we have CWD (current working directory) and CD (change directory) and not CWF of CF, my understanding is that `directory` is the technically correct term (especially when dealing with command line stuff).

But, I'm actually not too confident on that :stuck_out_tongue:
If you think people will be more familiar with the term `folder`, I'm fine changing it.

[1]: http://superuser.com/questions/169457/directory-vs-folder
[2]: http://superuser.com/questions/187900/what-is-the-difference-between-a-directory-and-a-folder
[3]: http://stackoverflow.com/questions/1191381/english-ui-terminology-directory-or-folder